### ISSUE TITLE
Increasing build speed

### ISSUE THREAD

**Issue by jdeluyck** [mmistakes/minimal-mistakes#2197]
I've been converting my WP blog to a Jekyll/MM one. I'm through the content changes to make everything work, but I'm getting massively annoyed by the build speeds.

After some initial cleanups (forgotten to use cached includes) I got it down from 3 minutes to 86 seconds, but still. Profiling the build comes up with:

```
Filename                                                                      | Count |     Bytes |   Time
------------------------------------------------------------------------------+-------+-----------+-------
_layouts/default.html                                                         |  1191 | 25797.46K | 27.326
_includes/archive-single.html                                                 |  1188 |  3517.86K | 24.523
_layouts/single.html                                                          |   363 | 11611.24K | 22.189
minimal-mistakes-jekyll-4.16.4/_includes/head.html                            |  1191 |  2020.59K | 18.526
minimal-mistakes-jekyll-4.16.4/_includes/seo.html                             |  1191 |  1216.90K | 16.496
_layouts/archive-taxonomy.html                                                |   753 |  1140.33K | 10.091
minimal-mistakes-jekyll-4.16.4/_includes/scripts.html                         |  1191 |  1140.64K |  6.315
_layouts/tags.html                                                            |     1 |  1430.28K |  6.069
```
for everything > 5 seconds.

I'm currently a bit puzzled what I might change/do to get stuff to build a bit quicker?

Sourcecode is on https://github.com/jdeluyck/kcoreorg-website

**Comment by mmistakes** [comment]
Unfortunately there's not much else you can do. As the amount of posts increases, so does the build. It's an issue with Jekyll in general and one of the reasons people move to Hugo.

Jekyll 4.0 has some nice improvements here that in my testing greatly reduce the amount of time sites take to build, so you could [use the beta gems](https://talk.jekyllrb.com/t/help-us-benchmark-jekyll/1629/17) if you aren't hosting on GitHub Pages.

The theme is also quite complex in a few places due to feature creep and offering flexibility to do many things. Downside is it increases the build time, so to fully fine tune things you'd need to build a custom fork and remove anything you don't use. Especially costly are LIquid `for` loops which you'll fine in the related posts component and the taxonomy archives.

Of interest might be my articles on how I use Jekyll. I've taken some drastic steps with my personal site to lower the build time which you might be able to take advantage of.

https://mademistakes.com/articles/using-jekyll-2016/
https://mademistakes.com/articles/using-jekyll-2017/

**Comment by jdeluyck** [comment]
I'm learning new things ;)

Stuck on: 
```
   jekyll-data was resolved to 1.0.0, which depends on
      jekyll (~> 3.3)
```

**Comment by mmistakes** [comment]
In my testing there were some nice speed improvements, but some things were broken. This was months ago so they may be fixed now, just something to be aware of if you're considering using it in production.
